### PR TITLE
Introduce gradle gradle spotless for java #771

### LINKED ISSUE #771: Provide gradle task to format sources
## Situation
In many code reviews source formatting is mentioned and the eclipse auto formatter  must be applied manually to these files. This is necessary but very cumbersome

## Wanted
Per command line there should be one call to format all source codes automatically - with our wanted setup.

## Solution
- We use a dedicated gradle plugin to format the code - e.g. `spotless` which can use eclipse formatter setup. see https://plugins.gradle.org/plugin/com.diffplug.spotless - if not suitable, there are others: see 
https://plugins.gradle.org/search?term=format

We can use
`./gradlew spotlessCheck` to verify

but also
`./gradlew spotlessApply` to automatically format code (currently only Java)

- We introduce also `prepareGitPush` task which dependsOn `spotlessApply`

### PR DESCRIPTION
**Pull request by de-jcup**
- added sechub eclipse formatter
- added importer order settings

This PR
- closes #771

### CODE REVIEW

**Comment by de-jcup** [comment]
@Jeeppler : It was necessary to apply the formatter to have a non failing build (spotless plugin is always automatically applied inside assemble task which is part of build). 

So there are extreme many files there. But for review it is pretty simple:

Just look at my first two commits
1. introduces spotless (exported our eclipse formatter settings and import order for SecHub to git repository)
2. introduces our custom `prepareGitPush` task
3. executed `prepareGitPush` and added all changes + commit + push

IMO you should 

A) check IDE formatter same as spotless
Simply checkout the branch inside your eclipse IDE, and make some spot checks by formatting some classes by eclipse formatter, maybe add some new lines etc. When the new lines are added a git delta is existing. After the IDE formatter has been passed, git status will not show any entries etc.

B) Check spotless works as thought
After this I would suggest to change some files again inside IDE (adding whitespaces e.g. tabs etc.)
Then execute 
`git status`
it must show some changes

After this execute
`./gradlew prepareGitPush`
in command line

Then execute 
`git status`
and it should show up no changes any more.

File: gradle/spotless.gradle
Comment (Jeeppler) [inline gradle/spotless.gradle:7]: What do you mean with `we do not use spotless here` do you mean for license files? 
What Bash file does "our solution"? Do you mean license headers?

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Please have a look at the comments. There might not be any code changes necessary.

Did I understand this properly, we apply Spotless to: Java, Groovy and dotfiles (files, where the filename starts with a `.` [dot])?

**Review by Jeeppler — APPROVED** [review]

File: gradle/spotless.gradle
Comment (de-jcup) [reply gradle/spotless.gradle:7]: see my next comment

File: gradle/spotless.gradle
Comment (de-jcup) [reply gradle/spotless.gradle:9]: Please read again:
```
currently we do not use spotless here, but our own bash solution - but spotless would be:
licenseHeaderFile rootProject.file('gradle/spotless.license')
```
We automatically apply our MIT headers via our bash script . This does handle it correctly and works well.
Spotless is also able to apply license header files automatically.
I just kept the example code inside, so it's easier when we want to switch (if ever). Also it describes that we have a bash solution here. So new team members will not accidently "enhance" the spotless setup.

Comment (Jeeppler) [reply gradle/spotless.gradle:9]: It this case please change the comment to:

~~~
// we currently do not use spotless to apply license headers. The license headers are added by our own bash solution:  `apply-headers.sh`.
// in case, we wanted to use spotless to apply license headers the next line needs to be uncommented.
// licenseHeaderFile rootProject.file('gradle/spotless.license')
~~~

Comment (de-jcup) [reply gradle/spotless.gradle:9]: As discussed, we just remove the comment.

**Comment by de-jcup** [comment]
We will go to commit 2, drop the comment (as requested). Wait for bigger PRs to be merged. After this we merge back develop to this branch and do the final global formatting and the PR will be merged as well.

**Pull request merged**
